sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala

```diff
@@ -225,6 +225,32 @@ class InsertSuite extends QueryTest with TestHiveSingleton with BeforeAndAfter
     checkAnswer(sql(selQuery), Row(5, 2, 3, 6))
   }
 
+  test("SPARK-37722: Escape dots in partition names") {
+    withTable("escapeDots") {
+      withSQLConf("hive.exec.dynamic.partition.mode" -> "nonstrict") {
+        sql("CREATE TABLE escapeDots (key int) PARTITIONED BY (value string);")
+        sql("INSERT INTO TABLE escapeDots VALUES (1, 'text.');")
+        sql("INSERT INTO TABLE escapeDots VALUES (2, '.');")
```

Comment:
can we test static partitions? `INSERT INTO TABLE escapeDots PARTITION(value = '.') VALUES (4)`